Eevee.Sleep.Bot/Handlers/ButtonClickedHandler.cs

```diff
@@ -0,0 +1,23 @@
+using Discord.WebSocket;
+using Eevee.Sleep.Bot.Controllers.Mongo;
+using Eevee.Sleep.Bot.Models;
+
+namespace Eevee.Sleep.Bot.Handlers;
+
+public static class ButtonClickedHandler {
+    public static async Task ChangeRoleButtonClicked(
```

Comment:
This will need to be able to differentiate between the command of "simply add" and "display certain role" as discussed in Discord dm yesterday.

The behavior here seems to be simply removing all tracked ones and add the selected one.

Reply:
It shows a list of roles that are being tracked when the command is launched, and nothing works without clicking on this.
Is this not enough functionality?
![image](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/assets/6148777/57f02bdd-ae11-4778-a9ca-0b2d4e2a3056)

Reply:
What I mean was clicking on the button can have 2 behaviors, so it needs differentiation:
1. Click on the button can be toggling the role on a user, so click on the button could result in either removing or adding the role to that user, depending on the user role ownership status
2. If this interaction is initiated with the "display certain role" command, all tracked roles will need to get removed from the command triggerer. Notice how 1. shouldn't have this behavior.

For 1., I think a better way to handle it is to split the toggling further into 2 commands, so we will have 3 commands that all showing this, but have different behavior.

Given the following, each command should have this behavior:
- User owns `@roleA` and `@roleB`
- User has `@roleA` on Discord only
- `@roleA` displays over `@roleB`

## `/role add`

When triggered, shows the roles that the triggerer owns, but not having it on Discord.

Using the conditions given, only `@roleB` should show to they can **add** `@roleB` on Discord to themselves. The tagging of `@roleA` shouldn't be affected. So after clicking on `@roleB`, they will have both `@roleA` and `@roleB` on Discord.

## `/role remove`

When triggered, shows the roles that the triggerer has, owns, but want to remove it on Discord.

Using the conditions given, only `@roleA` should show so they can **remove** `@roleA` on Discord. The ownership (our DB record) should get affected, meaning that the DB should still know the command triggerer has `@roleA` ownership and they can claim it back on Discord anytime.

## `/role display`

When triggered, shows all the roles that the triggerer owns, regardless if they have it on Discord. Selected role will be the **only** role they will have among the tracking roles after selecting it. An option of "none" might be needed.

Using the conditions given, both `@roleA` and `@roleB` should show, but after selecting either of them, they will only have either `@roleA` or `@roleB`, but not both.

Reply:
I see, it certainly didn't bother to delete the ones with lower display priority.
Understood.

Reply:
Add commands
[abfda0c](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25/commits/abfda0ca230b00a182aaf900f9d4e4b9ad21b67a)